app/frontend/src/features/profile/actions/MessageUserButton.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import MessageUserButton from "features/profile/actions/MessageUserButton";
+import { User } from "pb/api_pb";
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { groupChatsRoute, routeToGroupChat } from "routes";
+import { service } from "service";
+import users from "test/fixtures/users.json";
+import wrapper from "test/hookWrapper";
+import { MockedService } from "test/utils";
+
+const pushMock = jest.fn();
+const setErrorMock = jest.fn();
+const getDirectMessageMock = service.conversations
+  .getDirectMessage as MockedService<
+  typeof service.conversations.getDirectMessage
+>;
+jest.mock("react-router-dom", () => {
+  const pushMock = jest.fn();
+  return {
+    ...jest.requireActual("react-router-dom"),
+    useHistory: () => ({
+      push: pushMock,
+    }),
+  };
+});
```

Comment:
I would prefer to pass the (mocked) history to `hookWrapper`, but I encountered an issue where it is impossible to pass both history and state, because you need to use `<Router history={...}>`, but `createMemoryHistory()` doesn't support state in the entries. It is fixed in `history` v5, but `react-router-dom` uses v4.

Reply:
Do you mean if `history` had worked, you'd probably render the other component side-by-side and have it under `Router`? What's wrong with using `initialEntries` in `MemoryRouter`? Shouldn't it direct you to the right page with the state if you put them under the right `Route`s?